### ISSUE TITLE
No IssuerUpdated event after rotating the identity issuer certificate

### ISSUE THREAD

**Issue by anna-hope** [linkerd/linkerd2#8249]
### What is the issue?

Following [this guide](https://linkerd.io/2.11/tasks/manually-rotating-control-plane-tls-credentials/#rotating-the-identity-issuer-certificate) to rotate the identity issuer certificate, I replaced the certificate using `linkerd upgrade --identity-issuer-certificate-file=./issuer-new.crt --identity-issuer-key-file=./issuer-new.key | kubectl apply -f -`. 

I then ran `kubectl get events --field-selector reason=IssuerUpdated -n linkerd` to check the new certificate has been picked up. However, instead of the expected output as described in the guide, I got `No resources found in linkerd namespace.`

### How can it be reproduced?

- `step certificate create identity.linkerd.cluster.local issuer-new.crt issuer-new.key \
--profile intermediate-ca --not-after 87600h --no-password --insecure \
--ca ca-new.crt --ca-key ca-new.key` (note my `not-after` value)
- `linkerd upgrade --identity-issuer-certificate-file=./issuer-new.crt --identity-issuer-key-file=./issuer-new.key | kubectl apply -f -`
- `kubectl get events --field-selector reason=IssuerUpdated -n linkerd`

### Logs, error output, etc

`No resources found in linkerd namespace.`

### output of `linkerd check -o short`

```text
Linkerd core checks
===================


Status check results are √

Linkerd extensions checks
=========================


Status check results are √
```

### Environment

- Client: v1.22.5; Server: v1.21.5-gke.1805 
- Environment: GKE
- OS: macOS 12.3
- linkerd version: Client: stable-2.11.1; Server: stable-2.11.1

### Additional context

Running `kubectl get secret linkerd-identity-issuer -ojson -n linkerd |  jq '.data."crt.pem"' -r | base64 -d | step certificate inspect -`, I can verify that the certificate I generated matches the one on the server (public keys and serials match).

**Comment by olix0r** [comment]
@anna-hope Are you able to check your logs?

**Comment by anna-hope** [comment]
@olix0r Thank you for the reminder. I just tried regenerating and updating the certs, and lo and behold, a minute later the event showed up. Not sure why it didn't the first time. Please feel free to close the ticket.

**Comment by olix0r** [comment]
@anna-hope Weird! I'll close this but don't hesitate to open a new issue if you run into anything else unexpected :)

**Comment by sharmakhushboo** [comment]
Hi,

This issue still exists. Repro steps remain the same. Sometimes, after updating the issue cert, the IssuerUpdated event is not emitted. While updating the cert on multiple clusters, I observed that sometimes this event is emitted and sometimes its not. So, this to me does not seem like a reliable way to check whether the Issuer cert was updated or not.

However I was able to verify the Certificate updated using this command.
`kubectl get secret linkerd-identity-issuer -ojson -n linkerd | jq '.data."crt.pem"' -r | base64 -d | step certificate inspect -`
